### PR TITLE
[RLlib] JAXPolicy (working discrete-actions PPO prototype).

### PR DESCRIPTION
**Pull request by sven1977**
NOTE: Merge this PR here first for separation of concerns: https://github.com/ray-project/ray/pull/13091

JAXPolicy (working discrete-actions PPO prototype).

This PR adds:
- new JAXPolicy class
- PPOJAXPolicy class and loss
- Enhancements in JAXModelV2 to enable learning (get/set_weights).
- Working PPO compilation and LR schedule test case.
- Learning regression test for CartPole-v0 on PPO.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/a3c/a3c_torch_policy.py [hunk 1]
```diff
@@ -84,8 +84,9 @@ def _value(self, obs):
         return self.model.value_function()[0]
 
 
-A3CTorchPolicy = build_torch_policy(
+A3CTorchPolicy = build_policy_class(
```
Comment (ericl) [inline rllib/agents/a3c/a3c_torch_policy.py:87]: Can we do these kind of unrelated changes in a separate PR?

Comment (sven1977) [reply rllib/agents/a3c/a3c_torch_policy.py]: done

Comment (sven1977) [reply rllib/agents/a3c/a3c_torch_policy.py]: Also, review this one here first before proceeding: https://github.com/ray-project/ray/pull/13091

**Review by ericl — COMMENTED** [review]
Would be cool to also add JAX icon to the documentation (rllib.rst, rllib-algorithms.rst, rllib-toc.rst)

**Comment by sven1977** [comment]
A follow up PR will have documentation updates, including algo table with JAX sigils.

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.
